docs/reference/method/MongoDBModelCollectionInfo-getIdIndex.txt

```diff
@@ -0,0 +1,71 @@
+============================================
+MongoDB\\Model\\CollectionInfo::getIdIndex()
+============================================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Definition
+----------
+
+.. phpmethod:: MongoDB\\Model\\CollectionInfo::getIdIndex()
+
+   Returns information about the ``_id`` field index.
+
+   .. code-block:: php
+
+      function getIdIndex(): array
+
+Return Values
+-------------
+
+An array containing information on the ``_id`` index. This corresponds to the
+``idIndex`` field returned in the ``listCollections`` command reply.
+
+Examples
+--------
+
+.. code-block:: php
+
+   <?php
+
+   $info = new CollectionInfo([
+     'type' => 'view',
+     'name' => 'foo',
+     'idIndex' => [
+        'v' => 2,
+        'key' => ['_id' => 1],
+        'name' => '_id',
+        'ns' => 'test.foo',
+     ],
+   ]);
+
+   var_dump($info->getIdIndex());
```

Comment:
I'm not sure if these examples are really helpful. I believe we could optimise these by creating a collection first, then getting a `CollectionInfo` object for a collection and dumping from there. Right now, the example doesn't really clarify that the info here is server dependent.

Reply:
I concur. There's absolutely no reason to construct this directly, as that's not the intended use of this API.

It looks like these examples date back to [PHPLIB-278](https://jira.mongodb.org/browse/PHPLIB-278), which I signed off on. In the meantime, I'd suggest not adding any new examples and making a ticket to either replace all of these with integration examples _or_ remove them entirely and just point users to the corresponding enumeration method that would return these objects in the first place.

Reply:
I'll leave these examples in for consistency and have created PHPLIB-689 to track improvements.